web_poet/pages.py

```diff
@@ -63,10 +63,27 @@ def __init_subclass__(cls, skip_nonitem_fields: bool = False, **kwargs):
 
     async def to_item(self) -> ItemT:
         """Extract an item from a web page"""
+        validation_item = self._validate_input()
+        if validation_item is not None:
+            return validation_item
         return await item_from_fields(
             self, item_cls=self.item_cls, skip_nonitem_fields=self._skip_nonitem_fields
         )
 
+    @cached_method
+    def _validate_input(self) -> None:
+        """Run self.validate_input if defined."""
+        if not hasattr(self, "validate_input"):
+            return
+        if hasattr(self, "_validating_input"):
+            # We are in a recursive call, i.e. _validate_input is being called
+            # from _validate_input itself (likely through a @field method).
+            return
+        self._validating_input = True
```

Comment:
What do you think about making it protected (`__validating_input`)?

Reply:
The thing is we need it to be called in subclasses as well, and if we protect it things break in subclasses.

Are you suggesting it so that users do not accidentally use `_validate_input` for their own purposes, or for something else? If the former, maybe we can use some noise on the method, i.e. `_valudate_input_<some random characters>`.

Reply:
Hey! I was thinking about renaming the flag, not the method.